projects/helmet/helmet_project.py

```diff
@@ -42,9 +51,19 @@ def condition_func(self, total_results):
         person_model1 = self.mapping[person_model0][1]  # Mapping person from model1 to model0
         helmet_model0 = 1
 
-        return (any(box.cls == person_model0 for box in total_results[0].boxes)
-                and any(box.cls == helmet_model0 for box in total_results[0].boxes)
-                and any(box.cls == person_model1 for box in total_results[1].boxes))
+        has_person_model0 = any(box.cls == person_model0 for box in total_results[0].boxes)
```

Comment:
Just for my understanding. The `total_results` variable will contain all the boxes found in the frame, so if 2 persons, there will be 2 bounding boxes.

Looking at above logic, it will return `True` or `False` if at least one condition is met? And so below this code we check on width or height, so if at least 1 object is within the expected range, it will also accept the "non-valid" boxes.

Reply:
I think there is a confusion here, it will return `True` when ALL the conditions are met, otherwise `False`
E.g: total_results has to satisfy ALL:
  - At least 1 person in model 0 (using any()), 
  - At least 1 person in model 1, 
  - At least 1 helmet in model 1,
  - ALL person boxes in model 0 has greater width and height than min_width and min_height
  - ALL person boxes in model 1 has greater width and height than min_width and min_height

Because at the end I use `and` to verify so all conditions above must be met
`  if has_person_model0 and has_helmet_model0 and has_person_model1 and has_minimum_width_height_model0 and has_minimum_width_height_model1:
            return True`